integration-tests/test_client_options.py

```diff
@@ -17,3 +22,53 @@ def test_set_ansible_host_info(insights_client):
     ret = insights_client.run("--ansible-host=foo.example.com", check=False)
     assert "Could not update Ansible hostname" not in ret.stdout
     assert ret.returncode == 0
+
+
+def test_group(insights_client, tmp_path):
+    """Test --group option
+    This test generates an archive in offline mode and verifies if the group
+    specified in --group was created and packed
+    """
+
+    group_name = f"RHIQE.{uuid.uuid4()}.testing-group"
+    archive_tmp_name = f"test_archive_{uuid.uuid4()}.tar.gz"
+    archive_location = tmp_path / archive_tmp_name
+    tag_file = None
+
+    # Running insights-client in offline mode to generate archive
+    insights_client.run(
+        "--offline", f"--group={group_name}", f"--output-file={archive_location}"
+    )
+
+    with tarfile.open(archive_location, "r") as tar:
+        for m in tar.getmembers():
+            insights_archive_name = os.path.dirname(m.name)
+            if m.name == f"{insights_archive_name}/tags.json":
```

Comment:
rather than checking for the full path, wouldn't it be easier to simply check that the filename is simply `tags.json`? ie
```python
            if os.path.basename(m.name) == "tags.json":
```

Reply:
I thought of doing  so too but there are two tags.json in archive, i.e ../tags.json and ../data/tags.json

Reply:
there needs to be a different way then, because the current check matches both of them, so in the end whichever is found first is used for the test of the test

Reply:
ok I could use tar.getnames()[0] to get the dirname and check complete path.